src/test/app/LoadManifest_test.cpp

```diff
@@ -0,0 +1,239 @@
+                        "76463044022059979E116A61DE2CB0D9C5982F8D96E0A26AE9"
+                        "29526F851BF1949732186E089902204D03D8AA496D941F53DC"
+                        "51E9174E202264E5941EB72DDB3B45ED7116A226D00A770B65"
+                        "78616D706C652E6E6574701240527BCF45AB5496C80934A3E8"
+                        "4AE8E6DB9779F446B32CF00EF615DAAB41F692C31A044E9EE8"
+                        "D8F5D6F285624B86724EBE1E91CFC7CDA9BCBD61BAD2ED5086"
+                        "900F";
+
+// Seq 7, with different domain example.com
+static char const* m7 = "24000000077121ED1E79F744118D2602B8603EA6272DF04162"
+                        "495C785D3CEE2C4657D15E1CBEF8C87321033DE86C9CAD8DB9"
+                        "D0230305BCE9D074FB5337CFA51F5C175857331F378E5F22C2"
+                        "76473045022100839FC18FF980A64F13C5A07BFE429C7ADCD8"
+                        "6E671810F8BA066B3301A45263D202201E080E4B4C250A6D15"
+                        "C1C6BD6645ABF91B5E4CD6CE50E194B38E5FE50A50FC85770B"
+                        "6578616D706C652E6F726770124068268B6E6EE73CE2AFBCD7"
+                        "5D9C02D35361A19CCF71BFE6FA8DC890D454E49494A7C0D263"
+                        "B1FE43B3CD6DC2BC40141F4D5F3FCCF028D26530AADDCBA0C1"
+                        "024807";
+
+// Revocation
+static char const* mR = "24FFFFFFFF7121ED1E79F744118D2602B8603EA6272DF04162"
+                        "495C785D3CEE2C4657D15E1CBEF8C870124018F8520DC76445"
+                        "E17DC9232BC2575932574E0DC01C67A9305132B59E85AE46B0"
+                        "4312BAD3D7758F23EB918A847E7017A3DDCFB75AC6727A1DD7"
+                        "D67F5566907803";
+
+class LoadManifest_test : public beast::unit_test::suite
+{
+    void
+    testWithoutAmendment()
+    {
+        testcase("Without 'On Ledger Manifest' support");
+
+        using namespace test::jtx;
+        Env env(*this, supported_amendments() - featureOnLedgerManifests);
+
+        env(createManifest(env.master, m1), fee(env.current()->fees().reserve), ter(temUNKNOWN));
+        env.close();
+        env(updateManifest(env.master, m1), ter(temUNKNOWN));
+        env.close();
+        env(createManifest(env.master, mR), fee(env.current()->fees().reserve), ter(temUNKNOWN));
+        env.close();
+        env(updateManifest(env.master, mR), ter(temUNKNOWN));
+    }
+
+    void
+    testWithAmendment()
+    {
+        testcase("With 'On Ledger Manifest' support");
+
+        auto pk = parseBase58<PublicKey>(TokenType::NodePublic, mpk);
+        BEAST_EXPECT(pk);
+
+        using namespace test::jtx;
+        Env env(*this, supported_amendments());
+
+        auto checkSequence = [&](int expected)
+        {
+            auto sle = env.le(keylet::manifest(*pk));
+            return sle->getFieldU32(sfSequence) == expected;
+        };
+
+        // Updating manifest entry which isn't present fails
+        env(updateManifest(env.master, m1), ter(tecNO_ENTRY));
+        env.close();
```

Comment:
A few more negative test cases I'd like to see:

* `LoadManifest` with an invalid flag.
* Cases of `LoadManifest` with missing, empty, short (10 chars), and long (900 chars) `Manifest` fields.